lib/active_record/immutable.rb

```diff
@@ -0,0 +1,49 @@
+module ActiveRecord
+  module Immutable
+    extend ActiveSupport::Concern
+
+    FIELD_NAME_IDENTIFIER_IMMUTABLE = 'previous_version_id'.freeze
+
+    included do
+      def self.default_scope
+        where(%{
+          not exists (
+            select 1
+            from #{quoted_table_name} model
+            where model.#{FIELD_NAME_IDENTIFIER_IMMUTABLE} = #{quoted_table_name}.id
```

Comment:
What do you think about using the convention where we write the SQL commands as capital letters, on this case:
```
NOT EXISTS (
  SELECT 1
  FROM #{quoted_table_name} model
   WHERE model.#{FIELD_NAME_IDENTIFIER_IMMUTABLE} = #{quoted_table_name}.id
```